content/guides/java/inject-extract.md

```diff
@@ -2,8 +2,142 @@
 title: "Java: Inject and Extract"
 ---
 
-* When do you use Inject and Extract?
-* Carriers and Format Options
-* Injecting and Extracting HTTP
-* Injecting and Extracting TextMap
-* Injecting and Extracting Binary
+## Introduction
+
+In order to trace across process boundaries and RPC calls in distributed systems, span context needs to propagated over the wire. The Java Opentracing API provides two functions in the Tracer interface to do just that, `inject(SpanContext, format, carrier)` and `extract(format, carrier)`.
+
+## Format Options and Carriers
+The **format** parameter refers to one of the three standard encodings the OpenTracing API defines:
+
+1. `TEXT_MAP` where span context is encoded as a collection of string key-value pairs,
+2. `BINARY` where span context is encoded as an opaque byte array,
+3. `HTTP_HEADERS`, which is similar to `TEXT_MAP` except that the keys must be safe to be used as HTTP headers.
+
+The **carrier** is an abstraction over the underlying RPC framework. For example, a carrier for `TEXT_MAP` format is an interface that allows the tracer to write key-value pairs via `put(key, value)` method, while a carrier for Binary format is simply a ByteBuffer.
+
+## Injecting and Extracting HTTP
+The following section discusses injecting and extracting a span's context over HTTP requests.
```

Comment:
Could you add a bit more background information? Something like:

> When your service calls a downstream service, it's useful to pass down the `SpanContext`, so that `Span`s generated by this service could join the `Span`s from our service in a single trace. To do that, our service needs to `Inject` the `SpanContext` into the payload. On the other side of the connection, the downstream service needs then to `Extract` the `SpanContext` before it creates any `Spans`.
>
> Note that manually dealing with the `Inject` and `Extract` operations is not common in Java: there are OpenTracing instrumentation libraries that take care of this for you, such as the [java-web-servlet-filter](https://github.com/opentracing-contrib/java-web-servlet-filter) for the `extract` operation, or one of the following for the `inject` operation:
> * [java-okhttp](https://github.com/opentracing-contrib/java-okhttp)
> * [java-apache-httpclient](https://github.com/opentracing-contrib/java-apache-httpclient)
> * [java-asynchttpclient](https://github.com/opentracing-contrib/java-asynchttpclient)
> * [Spring's `RestTemplate`](https://github.com/opentracing-contrib/java-spring-web/tree/master/opentracing-spring-web)
>
> You are still encouraged to read on, to understand what happens behind the scenes.